compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -1021,7 +1021,7 @@ object Denotations {
      *  erasure (see i8615b, i9109b), Erasure takes care of adding any necessary
      *  bridge to make this work at runtime.
      */
-    def matchesLoosely(other: SingleDenotation)(using Context): Boolean =
+    def matchesLoosely(other: SingleDenotation, alwaysCompareParams: Boolean = false)(using Context): Boolean =
```

Comment:
I think this should be called `alwaysCompareTypes` since Type#matches can also check the result type if we're after erasure.